consensus/src/aggregation/engine.rs

```diff
@@ -0,0 +1,781 @@
+    rebroadcast_deadlines: PrioritySet<Index, SystemTime>,
+
+    // ---------- Journal ----------
+    /// Journal for storing acks signed by this node.
+    journal: Option<Journal<E, Activity<V, D>>>,
+    journal_name: String,
+    journal_write_buffer: usize,
+    journal_replay_buffer: usize,
+    journal_heights_per_section: u64,
+    journal_replay_concurrency: usize,
+    journal_compression: Option<u8>,
+
+    // ---------- Network ----------
+    /// Whether to send acks as priority messages.
+    priority_acks: bool,
+
+    /// The network sender and receiver types.
+    _phantom: PhantomData<(NetS, NetR)>,
+
+    // ---------- Metrics ----------
+    /// Metrics
+    metrics: metrics::Metrics<E>,
+}
+
+impl<
+        E: Clock + Spawner + Storage + Metrics,
+        P: PublicKey,
+        V: Variant,
+        D: Digest,
+        A: Automaton<Context = Index, Digest = D> + Clone,
+        Z: Reporter<Activity = Activity<V, D>>,
+        M: Monitor<Index = Epoch>,
+        TSu: ThresholdSupervisor<
+            Index = Epoch,
+            PublicKey = P,
+            Share = group::Share,
+            Identity = poly::Public<V>,
+        >,
+        NetS: Sender<PublicKey = P>,
+        NetR: Receiver<PublicKey = P>,
+    > Engine<E, P, V, D, A, Z, M, TSu, NetS, NetR>
+{
+    /// Creates a new engine with the given context and configuration.
+    pub fn new(context: E, cfg: Config<P, V, D, A, Z, M, TSu>) -> Self {
+        let metrics = metrics::Metrics::init(context.clone());
+
+        Self {
+            context,
+            automaton: cfg.automaton,
+            reporter: cfg.reporter,
+            monitor: cfg.monitor,
+            validators: cfg.validators,
+            namespace: cfg.namespace,
+            epoch_bounds: cfg.epoch_bounds,
+            window: cfg.window,
+            epoch: 0,
+            tip: 0,
+            safe_tip: SafeTip::default(),
+            digest_requests: FuturesPool::default(),
+            pending: BTreeMap::new(),
+            confirmed: BTreeMap::new(),
+            rebroadcast_timeout: cfg.rebroadcast_timeout,
+            rebroadcast_deadlines: PrioritySet::new(),
+            journal: None,
+            journal_name: cfg.journal_name,
+            journal_write_buffer: cfg.journal_write_buffer,
+            journal_replay_buffer: cfg.journal_replay_buffer,
+            journal_heights_per_section: cfg.journal_heights_per_section,
+            journal_replay_concurrency: cfg.journal_replay_concurrency,
+            journal_compression: cfg.journal_compression,
+            priority_acks: cfg.priority_acks,
+            _phantom: PhantomData,
+            metrics,
+        }
+    }
+
+    /// Runs the engine until the context is stopped.
+    pub fn start(mut self, network: (NetS, NetR)) -> Handle<()> {
+        self.context.spawn_ref()(self.run(network))
+    }
+
+    /// Inner run loop called by `start`.
+    async fn run(mut self, (net_sender, net_receiver): (NetS, NetR)) {
```

Comment:
nit: how do you feel about `sender/receiver`? I suppose that may get conflated with other channels?

Reply:
`tx`/`rx`/ seem like the "default" names to use in rust, which is fine enough as far as I'm concerned

Reply:
I left as-is for now but let me know if you think it's worth making any change now